app/components/npq_separation/primary_navigation_component.rb

```diff
@@ -0,0 +1,85 @@
+module NpqSeparation
+  class PrimaryNavigationComponent < ViewComponent::Base
+    attr_accessor :current_path, :sections
+
+    def initialize(current_path, structure:)
+      @current_path = current_path
+      @sections = mark_current(structure)
+    end
+
+    def build_sections
+      safe_join(
+        sections.map do |section|
+          tag.li(
+            link_to(section.name, section.path, class: "x-govuk-primary-navigation__link"),
+            class: class_names(
+              "x-govuk-primary-navigation__item",
+              "x-govuk-primary-navigation__item--current" => section == current_section,
+            ),
+          )
+        end,
+      )
+    end
+
+    def current_section
+      sections.find(&:current) || sections.first
+    end
+
+  private
+
+    def mark_current(structure)
+      structure.sections.each { |node| node.current = current_path.start_with?(node.prefix) }
+    end
+
+    class PrimaryNavigationStructure
```

Comment:
`PrimaryNavigationStructure` and `SubNavigationStructure` seem similar, if we extract those to let's say a separate `NavigationStructure` maybe we can use that in all the structures and they can live outside this file?